server/src/termdb.server.init.js

```diff
@@ -61,10 +61,10 @@ export function server_init_db_queries(ds) {
 		'anno_categorical',
 		'buildDate'
 	]
-	const sampleColumns = listTableColumns(cn, 'sampleidmap')
-	if (!sampleColumns.includes('type'))
-		//Fix datasets with missing columns on the fly
-		cn.prepare('alter table sampleidmap add column type character varying(100)').run()
+	//const sampleColumns = listTableColumns(cn, 'sampleidmap')
+	//if (!sampleColumns.includes('type'))
+	//Fix datasets with missing columns on the fly, requires write permission when opening the connection, not possible in readonly mode
```

Comment:
@airenzp Suggestion: add some helpers in `sjpp/utils/getDataset.js` to apply these alter and even create table statements, so when someone pulls a db file it can check for tables that need to be altered or added (for example, create sample_ancestry as needed, if the terms table exists in a db file).

Reply:
I did this instead to avoid the alter but still solve the issue of having to regenerate the dbs: https://github.com/stjude/proteinpaint/pull/1472/commits/f21f2b0bc3310f7acea55dd92c0a9ddb6a199796